### PR TITLE
initial pull request for code review

### PR DESCRIPTION
**Pull request by SviatAQA**

### CODE REVIEW

File: mvnw [hunk 1]
```diff
@@ -0,0 +1,316 @@
+#!/bin/sh
+# ----------------------------------------------------------------------------
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (pmichalskiGD) [inline mvnw]: do we need this file in repository?

File: mvnw.cmd [hunk 2]
```diff
@@ -0,0 +1,188 @@
+@REM ----------------------------------------------------------------------------
```
Comment (pmichalskiGD) [inline mvnw.cmd]: do we need this file in repository?

File: phonebook.iml [hunk 3]
```diff
@@ -0,0 +1,125 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pmichalskiGD) [inline phonebook.iml]: You should never push iml files into repository

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+			<artifactId>spring-boot-starter-test</artifactId>
+			<scope>test</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>org.springframework.boot</groupId>
+			<artifactId>spring-boot-starter-test</artifactId>
+			<version>RELEASE</version>
+			<scope>test</scope>
+		</dependency>
```
Comment (pmichalskiGD) [inline pom.xml]: this entry is duplicated

File: src/main/java/springcourse/phonebook/model/UserContact.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import javax.persistence.*;
+
+@Entity
+@Table(name = "contact")
+public class UserContact {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String contactName;
+    private String phoneNumber;
```
Comment (pmichalskiGD) [inline src/main/java/springcourse/phonebook/model/UserContact.java]: Shouldn't we keep list of phone numbers?
In task description it's specified that: 'PUT/{name}; phoneNumber - add phone to existing name' so I assume one contact can have several phone numbers

File: src/main/java/springcourse/phonebook/model/UserContact.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        this.contactName = firstName;
+    }
+
+    public String getPhoneNumber() {
+        return phoneNumber;
+    }
+
+    public void setPhoneNumber(String lastName) {
+        this.phoneNumber = lastName;
+    }
```
Comment (pmichalskiGD) [inline src/main/java/springcourse/phonebook/model/UserContact.java]: I highly encourage to use lombok: https://projectlombok.org/

File: src/main/resources/messages.properties [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#labels
```
Comment (pmichalskiGD) [inline src/main/resources/messages.properties]: are those properties used anywhere?

**Review by pmichalskiGD — COMMENTED** [review]
I need to say that I'm very pleased with overall code quality. Everything is well formatted and organized, and we only have couple of not needed files in this PR.

I found couple of bugs in the implementation:
- only one phone number per contact (also covered in comments)
- you can send 'id' in create contact request, and this allows to overwrite existing contact
- if you have contact with the same name, you will get 500 when you try to change phone number or delete it
- it's possible to set complete rubbish as phone number. It would be good to introduce some basic validation

File: mvnw (hunk 1 above)
Comment (SviatAQA) [reply mvnw]: Hello,
File was deleted from the repository and added to .gitignore

File: mvnw.cmd (hunk 2 above)
Comment (SviatAQA) [reply mvnw.cmd]: Hello,
File was deleted from the repository and added to .gitignore

File: phonebook.iml (hunk 3 above)
Comment (SviatAQA) [reply phonebook.iml]: Hello,
File was deleted from the repository and added to .gitignore

File: pom.xml (hunk 4 above)
Comment (SviatAQA) [reply pom.xml]: Fixed, My bad =(

File: src/main/java/springcourse/phonebook/model/UserContact.java (hunk 5 above)
Comment (SviatAQA) [reply src/main/java/springcourse/phonebook/model/UserContact.java]: Fixed.
I implemented it as a List of phone numbers.

File: src/main/java/springcourse/phonebook/model/UserContact.java (hunk 6 above)
Comment (SviatAQA) [reply src/main/java/springcourse/phonebook/model/UserContact.java]: Fixed.
Lombok library is added.

File: src/main/resources/messages.properties (hunk 7 above)
Comment (SviatAQA) [reply src/main/resources/messages.properties]: Fixed
File was removed

**Review by pmichalskiGD — APPROVED** [review]
Looks fine, ready to be merged.

Thank you for addressing all the comments!

**Pull request merged**